### ISSUE TITLE
Crash when deallocating canvas with <Sky /> element

### ISSUE THREAD

**Issue by flo-bit** [threlte/threlte#804]
[See a minimal example here](https://stackblitz.com/edit/nkthtn-wqdgqa?file=src%2Fexample%2FApp.svelte) toggle sky on, then toggle canvas off -> crash

Toggling the canvas/scene on or off without a sky component works without problems, when toggling the canvas off while it has a `<Sky />` I get the following runtime error:

```
TypeError: Cannot read properties of undefined (reading '0')
    at deallocateRenderTarget (chunk-EL5WXWEC.js?v=7bfaad47:15318:68)
    at WebGLCubeRenderTarget.onRenderTargetDispose (chunk-EL5WXWEC.js?v=7bfaad47:15277:5)
    at WebGLCubeRenderTarget.dispatchEvent (chunk-EL5WXWEC.js?v=7bfaad47:709:18)
    at WebGLCubeRenderTarget.dispose (chunk-EL5WXWEC.js?v=7bfaad47:2352:10)
    at @threlte_extras.js?v=7bfaad47:20042:50
    at run (chunk-7JRVFVIN.js?v=7bfaad47:28:10)
    at Array.forEach (<anonymous>)
    at run_all (chunk-7JRVFVIN.js?v=7bfaad47:34:7)
    at destroy_component (chunk-7JRVFVIN.js?v=7bfaad47:2148:5)
    at Object.destroy [as d] (Scene.svelte:5:39)
```

**Comment by MikeBellika** [comment]
I get the same crash report, except it happens whenever the app hot reloads

**Comment by grischaerbe** [comment]
fixed in #877 🎉
